core/lib/types/src/snapshots.rs

```diff
@@ -3,28 +3,47 @@ use zksync_basic_types::{L1BatchNumber, MiniblockNumber};
 
 use crate::{commitment::L1BatchWithMetadata, Bytes, StorageKey, StorageValue};
 
+/// Information about all snapshots persisted by the node.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct AllSnapshots {
+    /// L1 batch numbers for complete snapshots. Ordered by descending number (i.e., 0th element
+    /// corresponds to the newest snapshot).
     pub snapshots_l1_batch_numbers: Vec<L1BatchNumber>,
+    /// Same as `snapshots_l1_batch_numbers`, but for incomplete snapshots (ones in which some
+    /// storage log chunks may be missing).
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub incomplete_snapshots_l1_batch_numbers: Vec<L1BatchNumber>,
 }
 
-// used in dal to fetch certain snapshot data
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+/// Storage snapshot metadata. Used in DAL to fetch certain snapshot data.
+#[derive(Debug, Clone)]
 pub struct SnapshotMetadata {
+    /// L1 batch for the snapshot. The data in the snapshot captures node storage at the end of this batch.
     pub l1_batch_number: L1BatchNumber,
+    /// Path to the factory dependencies blob.
     pub factory_deps_filepath: String,
-    pub storage_logs_filepaths: Vec<String>,
+    /// Paths to the storage log blobs. Ordered by the chunk ID. If a certain chunk is not produced yet,
+    /// the corresponding path is `None`.
+    pub storage_logs_filepaths: Vec<Option<String>>,
+}
+
+impl SnapshotMetadata {
+    /// Checks whether a snapshot is complete (contains all information to restore from).
+    pub fn is_complete(&self) -> bool {
+        self.storage_logs_filepaths.iter().all(Option::is_some)
```

Comment:
Can we make this a column in db?

Reply:
this is denormalization and might result in inconsistent state

Reply:
I agree, but I think it would make it significantly easier to see the 'status' of the snapshot just by looking at the table

Reply:
currently, you'd have to write a query for that (there might be hundreds of chunks)

Reply:
but I should have definitely marked this comment as a 'nit'

Reply:
As we've discussed, normalization argument seems to win. We still have time to change the DB schema in the nearest future, so if we reconsider this decision, it could be implemented separately.